Adding new template views

siitinurliza opened the pull request.

siitinurliza committed:
"Add view: where-we-work-main"

siitinurliza committed:
Add Where We Work (Entry page) view

siitinurliza committed:
"Add view: imap-resource"

kaerumy commented:
@sitinurliza95 I would create and use the code for this for a new `country-view` instead and use the `where-we-work-view` as the main page template.

kaerumy commented:
```
  <browser:page
    name="front-page"
    for="*"
    class=".front_page.FrontPage"
    template="front_page.pt"
    permission="zope2.View"
    layer="imap.customizations.interfaces.IImapCustomizationsLayer"
    />
```

Renaming isn't that complicated though, see configure.zcml in the views directory.
Just need to change the respective class and template variables to the new names.

In this example: 

`.front_page` is `front_page.py` filename that can be changed.
`FrontPage` is the class name in `front_page.py` which can be changed 
`front_page.pt` is just the filename and be changed to the new renamed template eg. `country-view.pt`

siitinurliza committed:
Renamed files and add Resource page view

siitinurliza committed:
Added css files and updated imap-resource template and front-page template

siitinurliza committed:
Added css files for document headings and Resource page

siitinurliza committed:
Added CSS files for Country page view and updated CSS files for Resource page view

siitinurliza committed:
Changes to be committed:
	modified:   src/imap/customizations/theme/css/theme.css
	modified:   src/imap/customizations/theme/css/theme.css.map
	modified:   src/imap/customizations/theme/css/theme.min.css
	modified:   src/imap/customizations/theme/css/theme.min.css.map
	modified:   src/imap/customizations/theme/scss/theme.scss
	modified:   src/imap/customizations/views/front_page.pt

siitinurliza committed:
"Add content_type: Person"

siitinurliza committed:
"Add content_type: Internet censorship event"

siitinurliza committed:
Added front page layout, remove content type Person and other minor changes

siitinurliza committed:
Added fields to internet censorship event

siitinurliza committed:
Updated iMAP resource view with TAL statements (incomplete)

siitinurliza commented:
Issue: Unable to view all SDG goals (and other similar fields) related to a Resource

For example in `imap_resource.py`
```
   def sdg_title(self):

        factory = getUtility(IVocabularyFactory,
                             'sinar.miscbehavior.SDGGoals')

        vocabulary = factory(self.context.SDG_goals)
        
        for term in vocabulary:
            if self.context.SDG_goals:
                return term.title
            else:
                return None
```

and `imap_resource.pt`

```
<p tal:content="view/sdg_title" id="resource-card-top-text">Encourage and promote effective public, public-
                      private,
                      and civil society partnerships, building on the experience and resourcing
                      strategies of partnerships.</p>
```

This returns only the first SDG goal related to the Resource as below, even though there is more than 1 SDG goal in the data:

![image](https://github.com/Sinar/imap.customizations/assets/86417001/22286d21-8d93-409f-b335-d15d80a3eb21)

![image](https://github.com/Sinar/imap.customizations/assets/86417001/005c9514-354d-43e5-9a7d-cd87c2e4c831)

siitinurliza committed:
Updated iMAP resource view with TAL statements (incomplete)

siitinurliza committed:
Updated header and footer in main template

siitinurliza committed:
Updated header and footer in main template

siitinurliza commented:
Pending issues:

Front page, header, footer

- [ ] Can't activate javascript related objects e.g. carousel (for the banner), nav and tabs (for the events)
- [ ] Unable to show listing of resources and news 
- [ ] Unable to show varying logos of countries and partners

Resource page
- [ ] Unable to view all SDG goals/ countries/ similar metadata related
- [ ] Author names and tags come with brackets `('Siti Khadijah', 'Yusoof Baki')`
- [ ] Unable to show organizations related using `tal repeat`
- [ ] Need to add key findings as behavior (?)

Country page
- [ ] Unable to show listing of resources and news
- [ ] Unable to show varying logos of countries and partners

Others
- [ ] Adding view for internet censorship events
- [ ] Adding view for listing pages with filters
